pinot-common/src/main/java/org/apache/pinot/common/config/provider/TableCache.java

```diff
@@ -543,13 +549,44 @@ private TableConfigInfo(TableConfig tableConfig) {
     }
   }
 
-  private static class SchemaInfo {
+  public static class SchemaInfo {
     final Schema _schema;
     final Map<String, String> _columnNameMap;
+    int _dimensionFieldSpecsCount;
+    final int _dateTimeFieldSpecsCount;
+    final int _metricFieldSpecsCount;
```

Comment:
Do we need to cache these values? These values can be calculated on the fly because schema is always loaded in memory, and the computation is very cheap.
We can probably keep this class as is, and extract the field count info from the schema when the API is invoked

Reply:
In order to know the dimension fields count without virtual column, we can deduct 3, or only cache `numDimensionFields`